From 68685f9289439982855eaa5fffa4dad3b92da623 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Wed, 27 Jun 2012 16:12:22 +0800
Subject: ENGR00215041-3 MX6 SabreSD:Correct camera and audio mclk freq

This patch corrects camera mclk and audio mclk frequency
to be 24MHz to align with 24MHz osc clock.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit 00ea48ba38940c3a908f9a5d2e72ae285a221329)
---
 arch/arm/mach-mx6/board-mx6q_sabresd.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabresd.c b/arch/arm/mach-mx6/board-mx6q_sabresd.c
index 7f8add9..cfd1cb3 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabresd.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabresd.c
@@ -372,7 +372,7 @@ static int mxc_wm8958_init(void)
 		return PTR_ERR(clko);
 	}
 	/* both audio codec and comera use CLKO clk*/
-	rate = clk_round_rate(clko, 22000000);
+	rate = clk_round_rate(clko, 24000000);
 
 	wm8958_data.sysclk = rate;
 	clk_set_rate(clko, rate);
@@ -412,7 +412,7 @@ static int mxc_wm8962_init(void)
 		return PTR_ERR(clko);
 	}
 	/* both audio codec and comera use CLKO clk*/
-	rate = clk_round_rate(clko, 22000000);
+	rate = clk_round_rate(clko, 24000000);
 	clk_set_rate(clko, rate);
 
 	wm8962_data.sysclk = rate;
@@ -523,7 +523,7 @@ static void mx6q_csi0_io_init(void)
 }
 
 static struct fsl_mxc_camera_platform_data camera_data = {
-	.mclk = 22000000,
+	.mclk = 24000000,
 	.mclk_source = 0,
 	.csi = 0,
 	.io_init = mx6q_csi0_io_init,
-- 
1.7.9.5

